evalml/pipelines/components/transformers/samplers/oversamplers.py

```diff
@@ -0,0 +1,59 @@
+from evalml.pipelines.components.transformers.samplers.base_sampler import (
+    BaseOverSampler
+)
+from evalml.utils.woodwork_utils import infer_feature_types
+
+
+class SMOTESampler(BaseOverSampler):
+    """SMOTE Oversampler component. Works on numerical datasets only. This component is only run during training and not during predict."""
+    name = "SMOTE Oversampler"
+    hyperparameter_ranges = {}
+
+    def __init__(self, sampling_ratio=0.25, k_neighbors=5, n_jobs=-1, random_seed=0, **kwargs):
+        super().__init__("SMOTE",
+                         sampling_ratio=sampling_ratio,
+                         k_neighbors=k_neighbors,
+                         n_jobs=n_jobs,
+                         random_seed=random_seed,
+                         **kwargs)
+
+
+class SMOTENCSampler(BaseOverSampler):
+    """SMOTENC Oversampler component. Uses SMOTENC to generate synthetic samples. Works on a mix of nomerical and categorical columns.
+       Input data must be Woodwork type, and this component is only run during training and not during predict."""
+    name = "SMOTENC Oversampler"
+    hyperparameter_ranges = {}
+
+    def __init__(self, sampling_ratio=0.25, k_neighbors=5, n_jobs=-1, random_seed=0, **kwargs):
+        self.categorical_features = None
+        super().__init__("SMOTENC",
+                         sampling_ratio=sampling_ratio,
+                         k_neighbors=k_neighbors,
+                         n_jobs=n_jobs,
+                         random_seed=random_seed,
+                         **kwargs)
+
+    def _get_categorical(self, X):
+        X = infer_feature_types(X)
+        self.categorical_features = [i for i, val in enumerate(X.types['Logical Type'].items()) if str(val[1]) == 'Categorical']
```

Comment:
For this, we need either a list of indices or a mask of booleans. I opted for the list of indices, and this was the easiest way to get the indices rather than the column names

Reply:
Thanks for explaning!